user/languages/_includes/modern-c-cpp-toolchain.md

```diff
@@ -0,0 +1,130 @@
+### Modern C/C++ and Toolchain Versioning
+
+If you need a more modern toolchain or more granular control over your tools, you will have to handle that yourself. Fortunately, the community has already whitelisted the [repositories](https://github.com/travis-ci/apt-source-whitelist/blob/master/ubuntu.json) and [packages](https://github.com/travis-ci/apt-package-whitelist/blob/master/ubuntu-precise) necessary for most such upgrades.
+
+#### GCC on Linux
+
+Ubuntu 12.04 shipped with GCC 4.6.3 and Ubuntu 14.04 shipped with GCC 4.8.2. If you need a more recent version (if you are working with C++11, you *must* do this; support for C++11 was not finished until GCC 4.9), you will want to follow these directions:
```

Comment:
1. Is that actually possible? The 12.04 images have already been released and mirrored thoroughly extensively. Even in the scenario where the base packages are bumped, they wouldn't just drop them into 12.04 itself, but rather release a new version, [like they did with 12.04.3](https://wiki.ubuntu.com/PrecisePangolin/ReleaseNotes/ChangeSummary/12.04.3).

1. I'm loath to add another section in its entirety to address C++11 support specifically, since (1) this entire PR's purpose is to add a section titled "Modern C/C++ and Toolchain Versioning" and (2) there really isn't much to say about supporting C++11 *specifically* besides what's here in this parenthetical. Supporting C++14 and C++17 are different stories entirely.

    Also, @BanzaiMan I suggested above in response to @plaindocs that the parenthetical be replaced with the following - thoughts (this also addresses the lack of clarity w.r.t. "this")?

    > (note: C++11 is feature-complete in 4.8.1, but [support for `<regex>` does not exist until 4.9](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=53631)

1. The "follow these directions" language can be adjusted, yes.